src/context/country-context.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, createContext } from "react";
+
+const defaultCountry = {
+    code: null,
+    name: null,
+}
+
+const CountryContext = createContext({
+    country: defaultCountry,
+    setCountry: () => {},
+});
+
+const CountryProvider = ({ children }) => {
+
+    const [country, _setCountry] = useState(defaultCountry);
+
+    const setCountry = newCountry => {
+        _setCountry(newCountry);
+    };
```

Comment:
This is actually redundant, `_setCountry` can be renamed to `setCountry` above and this can be removed. It exists becuase I thought more logic would have to happen here but that turned out not to be the case.